### PR TITLE
MTDSA-25367: Reject requests for TY 25-26 onwards for v4 and v5 endpoints (RULE_TAX_YEAR_NOT_SUPPORTED error)

### PR DESCRIPTION
**Pull request by GazalNebHMRC**

### CODE REVIEW

File: resources/public/api/conf/4.0/foreign_property_accounting_adjustments_submit.yaml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 post:
   summary: Submit Foreign Property Accounting Adjustments
   description: |
-    This endpoint allows the user to provide accounting adjustments against a specified Business Source Accounting 
-    Summary quoting its unique identifier. The calculation ID quoted must be for a Foreign Property Business, and it 
-    must not have been adjusted previously. Only data for one property business either foreign property (FHL EEA) or 
-    foreign property (Non FHL) should be included in any submission. A National Insurance number and calculation ID must
-    be provided.
+    This endpoint enables you to provide accounting adjustments against a Business Source Accounting Summary specified by its calculation ID.
+    The calculation ID quoted must be for a UK property business, and it must not have been adjusted previously.
```
[the remark this answers is not in the document]
Comment (GazalNebHMRC) [reply resources/public/api/conf/4.0/foreign_property_accounting_adjustments_submit.yaml:5]: Good spot, it seems this was in the confluence spec like this https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?pageId=872186888for all foreign property endpoints. I will update in the code and the confluence specs as well

**Review by FergusHMRC — DISMISSED** [review]

File: app/v4/controllers/ListBsasController.scala [hunk 2]
```diff
@@ -40,6 +40,8 @@ class ListBsasController @Inject() (val authService: EnrolmentsAuthService,
     extends AuthorisedController(cc)
     with Logging {
 
+  val endpointName = "list-bsas"
+
   implicit val endpointLogContext: EndpointLogContext =
     EndpointLogContext(controllerName = "ListBsasController", endpointName = "listBsas")
```
Comment (FergusHMRC) [inline app/v4/controllers/ListBsasController.scala:46]: could the endpointName val be used here to reduce future changes required if name changed?

File: app/v4/controllers/RetrieveForeignPropertyBsasController.scala [hunk 3]
```diff
@@ -30,16 +30,20 @@ import javax.inject.{Inject, Singleton}
...
+    cc: ControllerComponents,
+    val idGenerator: IdGenerator
+)(implicit ec: ExecutionContext, appConfig: AppConfig)
     extends AuthorisedController(cc)
     with Logging {
 
+  val endpointName = "retrieve-foreign-property-bsas"
+
   implicit val endpointLogContext: EndpointLogContext =
     EndpointLogContext(controllerName = "RetrieveForeignPropertyBsasController", endpointName = "retrieve")
```
Comment (FergusHMRC) [inline app/v4/controllers/RetrieveForeignPropertyBsasController.scala:48]: the endpointNames dont match here?

File: app/v4/controllers/ListBsasController.scala (hunk 2 above)
Comment (GazalNebHMRC) [reply app/v4/controllers/ListBsasController.scala]: These names were like this in the main branch. I haven't made these changes, but we can discuss this with others to check why the endpoint names don't match

File: test/v4/controllers/ListBsasControllerSpec.scala [hunk 4]
```diff
@@ -143,7 +144,7 @@ class ListBsasControllerSpec
       incomeSourceType = Some(typeOfBusiness)
     )
 
-    val response: ListBsasResponse[BsasSummary] = ListBsasResponse(
+    val response: ListBsasResponse[BsasSummary] = models.response.listBsas.ListBsasResponse(
```
Comment (FergusHMRC) [inline test/v4/controllers/ListBsasControllerSpec.scala:147]: any reason why this has an explicit call when ListBsasResponse is directly imported above?

**Review by FergusHMRC — DISMISSED** [review]

File: app/v6/selfEmploymentBsas/submit/model/request/SubmitSelfEmploymentBsasRequestBody.scala [hunk 5]
```diff
@@ -14,26 +14,18 @@
...
 
-package v3.models.request.triggerBsas
+package v6.selfEmploymentBsas.submit.model.request
 
-import play.api.libs.json.Json
-import shared.utils.UnitSpec
-import v3.fixtures.TriggerBsasRequestBodyFixtures._
+import play.api.libs.json.OWrites
+import shared.utils.JsonWritesUtil
+import v5.selfEmploymentBsas.submit.def1.model.request.Def1_SubmitSelfEmploymentBsasRequestBody
```
Comment (juamalBlackman) [inline app/v6/selfEmploymentBsas/submit/model/request/SubmitSelfEmploymentBsasRequestBody.scala:21]: should be v6 import here instead

**Review by FergusHMRC — APPROVED** [review]

**Pull request merged**